src/NumericalAlgorithms/Spectral/Mesh.hpp

```diff
@@ -48,6 +49,39 @@ class Mesh {
  public:
   static constexpr size_t dim = Dim;
 
+  struct Extents {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "The number of collocation points per dimension"};
+  };
+
+  struct CfoBasis {
+    static std::string name() noexcept { return "Basis"; };
+    using type = Spectral::Basis;
+    static constexpr Options::String help = {
+        "The choice of spectral basis to compute the collocation points"};
+  };
+
+  struct CfoQuadrature {
+    static std::string name() noexcept { return "Quadrature"; };
+    using type = Spectral::Quadrature;
+    static constexpr Options::String help = {
+        "The choice of quadrature to compute the collocation points"};
+  };
+
+  using options = tmpl::list<Extents, CfoBasis, CfoQuadrature>;
+
+  static constexpr Options::String help =
+      "Holds the number of grid points, basis, and quadrature in each"
+      "direction of the computational grid."
+      "A mesh encapsulates all information necessary to construct the"
+      "placement of grid points in the computational domain. It does so "
+      "through a choice of basis functions, quadrature and number of points "
+      "\\f$N\\f$ "
+      "in each dimension. The grid points are the associated collocation "
+      "points and can be"
+      "obtained from Spectral::collocation_points(const Mesh<1>&)";
```

Comment:
Drop the last sentence (or at least the part starting from "and can be").  It's not useful to someone creating a Mesh in an input file.